bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/GetParamsetParser.java

```diff
@@ -61,8 +61,7 @@ public Void parse(Object[] message) throws IOException {
                 boolean isHmSenMdirNextTrans = dpInfo.getName().equals("NEXT_TRANSMISSION")
                         && (deviceType.startsWith("HM-Sen-MDIR-O") || deviceType.startsWith("HM-Sen-MDIR-WM55"));
                 if (!isHmSenMdirNextTrans) {
-                    if (dpInfo.getAddress().contains(":M_")
-                            && channel.getDevice().getHmInterface() == HmInterface.HMIP) {
+                    if (dpInfo.toString().contains(":M_") && channel.getDevice().getHmInterface() == HmInterface.HMIP) {
```

Comment:
Looking at what it actual checks. I think the check should be on the enum. Because the first character of enum is what is printed in the toString and what is check for here.
```suggestion
                    if (dpInfo.getParamsetType() == HmParamsetType.MASTER && channel.getDevice().getHmInterface() == HmInterface.HMIP) {
```

Reply:
Thanks for the hint. It's definitely better. No idea how I could have missed the enum :-) . Changed it and added another small commit to the same method.